fluere-plugin/src/util.rs

```diff
@@ -10,19 +10,23 @@ pub fn home_cache_path() -> PathBuf {
 
     let path_base = match sudo_user {
         Ok(user) => {
-            // If SUDO_USER is set, construct the path using the user's home directory
-            let user_home = format!("/home/{}", user);
-            Path::new(&user_home).join(".cache")
+            // on macOS just return the cache_dir()
+            if env::consts::OS == "macos" {
+                cache_dir().expect("Could not determine the home directory")
+            } else {
+                // If SUDO_USER is set, construct the path using the user's home directory
+                let user_home = format!("/home/{}", user);
+                Path::new(&user_home).join(".cache")
+            }
         }
         Err(_) => {
-            // If not running under sudo, just use the config_dir function as before
+            // If not running under sudo, just use the cache_dir function as before
             cache_dir()?
```

Comment:
error[E0277]: the `?` operator can only be used in a function that returns `Result` or `Option` (or another type that implements `FromResidual`)
  --> fluere-plugin/src/util.rs:24:24
   |
7  | pub fn home_cache_path() -> PathBuf {
   | ----------------------------------- this function should return `Result` or `Option` to accept `?`
...
24 |             cache_dir()?
   |                        ^ cannot use the `?` operator in a function that returns `PathBuf`
   |
   = help: the trait `FromResidual<Option<Infallible>>` is not implemented for `PathBuf`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `fluere_plugin` (lib) due to previous error